bson/__init__.py

```diff
@@ -284,10 +284,10 @@ def _get_object_size(data: Any, position: int, obj_end: int) -> Tuple[int, int]:
     except struct.error as exc:
         raise InvalidBSON(str(exc)) from None
     end = position + obj_size - 1
-    if data[end] != 0:
-        raise InvalidBSON("bad eoo")
     if end >= obj_end:
         raise InvalidBSON("invalid object length")
+    if data[end] != 0:
+        raise InvalidBSON("bad eoo")
```

Comment:
Was this another bug fix?

Reply:
Yes, to avoid an `IndexError` instead of `InvalidBSON`.